students/CarolF/module05/MailRoom_3_CFarris.py

```diff
@@ -0,0 +1,129 @@
+    print("\nDear {},".format(donorName)) #added newlines to improve readability
+    print("Thank you for your generous contribution of ${} ".format(newContribution))
+    print("Your contribution is greatly appreicated and will be used to grow our operations and will reinforce our commitment to innovation.")
+    print("Sincerely,")
+    print("The Entire Team at PyCert.com\n")
+
+
+#still need to call and improve formattting.
+def createReportData():
+    print('{:<15}'.format("Donor Name"), '{:<20}'.format('|  Total Given  |'),
+                 '{:>5}'.format('Number of Gifts  |'),'{:>20}'.format('Average Gift  |')) 
+    print('_'*77)
+    dataForDonationReport = []
+    for key in donorDict:
+        donorNumTotal = sum(donorDict[key])
+        numbGifts = (len(donorDict[key]))
+        donorAverage = "$ {0:.2f}".format(float((sum(donorDict[key]))/numbGifts))
+        dataForDonationReport.append([donorNumTotal,key,numbGifts,donorAverage]) 
+    
+    sortedReport = sorted(dataForDonationReport)
+    ascendingReport =sortedReport[::-1]
+    
+    for donor in ascendingReport:
+        donorTotal = "$ {0:.2f}".format(donor[0]) 
+        print('{:<20}'.format(donor[1]),'{:<15}'.format(donorTotal),
+                '{:>5}'.format(donor[2]),'{:>25}'.format(donor[3]))   
+    print('_'*77)
+    print('\n') 
+    print('Please indicate your next action')   
+
+
+def sendLettersToAll():
+    """Prints to local directory thank you letters to all contributors."""      
+    print("Get Ready for some thank yous!")
+    for key in donorDict:
+        with open(f'{key}.txt', 'w')as f:
+            f.write(f'Dear {key},\n')
+            f.write(f'Thank you for your generous support of ${sum(donorDict[key])} to our company. ')
+            f.write(f'It is through the support of contributors such as yourself that we\n')
+            f.write(f'are able to  bring our innovative spirit into the marketplace. ' )
+            f.write(f'Please enjoy the enclosed shrubbery one of our team members grew\nfor you! ')
+            f.write(f'With our deepest appreciation,\n Our team at PyCert')
+
+
+def printThankYouLetter(): 
+    foundDonors = findDonorName()
+    foundDonation = retrieveDonationAmount(foundDonors)
+    printLetter(foundDonors, foundDonation)
+
+
+
+if __name__ == "__main__" :
+  """Only the action list is listed as that method routes the user to the desired action  """
+
+validSelection = False
+while validSelection == False: ##Ensure user enters coorect selection 
+    try: 
+        for key, value in usersChoice.items():    print('%s: %s' % (key, value))
+        response = int(input("Please type a numeric selection from the above actions >"))  
+    except ValueError:
+        print("\nInput must be an integer. Try again")
+    else: 
+        if response != 1 and response != 2 and response != 3 and response != 4:
+            print("Invalid entry. Expected a '1', '2', '3', or '4' ")
+        else:
```

Comment:
This would be better as a list: 

    response_list = [1,2,3,4]
    if response not in response_list:
        do things